be/src/olap/rowset/segment_v2/binary_prefix_page.cpp

```diff
@@ -215,7 +216,10 @@ Status BinaryPrefixPageDecoder::next_batch(size_t* n, vectorized::MutableColumnP
     // read and copy values
     for (size_t i = 0; i < max_fetch; ++i) {
         dst->insert_data((char*)(_current_value.data()), _current_value.size());
-        _read_next_value();
+        if (_next_ptr >= _footer_start) {
```

Comment:
This is maybe useless, because it should not happen. If this case happens, I think there is a fatal bug. BE should log fatal here.

Reply:
it will always try to decode the entry at `_footer_start `(a invalid position) after it read the last value of the this page. We found this bug after https://github.com/apache/doris/pull/23425 makes `Status::Corruption` print stacktrace and we found that be.INFO are full of stacktrace of `Status::Corruption` from `_decode_value_lengths`

Reply:
While `i` < `max_fetch`, `_next_ptr` should not reach the position of `_footer_start`？
In the other word, while `_next_ptr` reached the position of `_footer_start`, `max_fetch` should be zero, or `i` would be equal with `max_fetch`